src/cpu/flexcpu/flexcpu_thread.cc

```diff
@@ -913,6 +948,32 @@ FlexCPUThread::onIssueAccessed(weak_ptr<InflightInst> inst)
     inst_ptr->traceData(nullptr);
 #endif
 
+    if (inst_ptr->staticInst()->isLastMicroop()) {
+        DPRINTF(FlexCPUThreadEvent, "Releasing MacroOp after decoding final "
+                                    "microop\n");
+        curMacroOp = StaticInst::nullStaticInstPtr;
+    }
+
+    const bool interrupt_detected = _cpuPtr->checkInterrupts(this);
```

Comment:
Why is this needed at issue time?

Reply:
This is for optimization in the same way branch prediction ends early if we think we can know earlier that an interrupt occurred. We kill any issue logic when an interrupt occurs so that the CPU spends less time issuing, when an interrupt is going to squash all of it anyway on the next time a commit occurs.